lighthouse-core/audits/image-size-responsive.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @license Copyright 2020 Sebastian Kreft All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+/**
+ * @fileoverview Checks to see if the size of the visible images used on
+ *   the page are large enough with respect to the pixel ratio. The
+ *   audit will list all visible images that are too small.
+ */
+'use strict';
+
+const Audit = require('./audit.js');
+const URL = require('../lib/url-shim.js');
+const i18n = require('../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on the size of visible images on the page. This descriptive title is shown to users when all images have correct sizes. */
+  title: 'Displays images with correct size',
+  /** Title of a Lighthouse audit that provides detail on the size of visible images on the page. This descriptive title is shown to users when not all images have correct sizes. */
+  failureTitle: 'Displays images with incorrect size',
+  /** Description of a Lighthouse audit that tells the user why they should maintain the correct aspect ratios for all images. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Image natural dimensions should be proportional to the display size and the ' +
+    'pixel ratio. [Learn more](https://web.dev/image-size-responsive).',
+  /**  Label for a column in a data table; entries in the column will be a string representing the displayed size of the image. */
+  columnDisplayed: 'Displayed size',
+  /**  Label for a column in a data table; entries in the column will be a string representing the actual size of the image. */
+  columnActual: 'Actual size',
+  /**  Label for a column in a data table; entries in the column will be a string representing the expected size of the image. */
+  columnExpected: 'Expected size',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+const TOLERANCE = 0.75;
+
+const ALLOWABLE_OFFSCREEN_X = 100;
```

Comment:
any guidance as to where these came from/why they were chosen or how to update them in the future? I suspect the answer is just "it's what offscreen images did and that has no explanation" 😆 but even that would be useful to capture so we update both  at the same time :)

Reply:
You are right that I took it directly from the `OffscreenImages` audit. Now, after second though I think that maybe we should get rid of the `ALLOWABLE_OFFSCREEN_X`, and `ALLOWABLE_OFFSCREEN_Y`. Instead we should just check the image is within (intersects) the area [0, innerWidth] x [0, innerHeight], or maybe [0, innerWidth] x [0, innerHeight + OFFSET] or maybe [0, innerWidth] x [0, innerHeight * FACTOR] (FACTOR=1.5?).

WDYT?

Reply:
Yeah SGTM! 👍 

I'd be on board for narrowest possible set to ensure no conflicting advice with offscreen images. (i.e. we shouldn't have a situation where offscreen tells you to lazy load and this tells you to preload)

Reply:
I changed it to just check the area [0, innerWidth] x [0, innerHeight]